### PR TITLE
Merge PR #274 to 1.12 Branch

### PR DESCRIPTION
**Pull request by fortnern**
Fix problems with vlens and refs inside compound using H5VLget_file_type()

### CODE REVIEW

**Comment by fortnern** [comment]
I don't know why this PR is including the merge for #270 , which was already merged to hdf5_1_12

**Comment by fortnern** [comment]
Ok I think it's fixed now.

File: src/H5Tprivate.h
Comment (byrnHDF) [inline src/H5Tprivate.h]: TYPO, I found and fixed it in my whitespace PR#282

Comment (fortnern) [reply src/H5Tprivate.h]: Thanks

Comment (fortnern) [reply src/H5Tprivate.h]: I'll just wait until that one is merged.

**Comment by fortnern** [comment]
So turns out a different mistake was the culprit - I forgot to "add" a fix to the original merge commit that was present in my working directory. I think it makes sense to just let the other issue be fixed when Allen's change is merged to 1.12 since it isn't breaking anything right now.

**Review by lrknox — APPROVED** [review]
Approved provided the correction to line 44 of H5Tprivate.h is preserved in hdf5_1_12.

**Comment by byrnHDF** [comment]
> Approved provided the correction to line 44 of H5Tprivate.h is preserved in hdf5_1_12.

Yes, that has to be fixed in this PR because it doesn't exist yet for me to fix.

**Comment by lrknox** [comment]
You fixed it in PR #282 which has been merged to develop, but I guess not to hdf5_1_12?

File: src/H5Tprivate.h
Comment (lrknox) [inline src/H5Tprivate.h]:
```suggestion
#define H5T_GET_FORCE_CONV(T)       ((T)->shared->force_conv)
```

Comment (byrnHDF) [reply src/H5Tprivate.h]: That only exists in this PR, until it is merged it is not in 1.12. Neal needs to make the change for it to appear in this PR. 
OR
Merge this PR then go back and edit.

**Comment by fortnern** [comment]
I figured it would save Allen the headache of merge conflicts if I didn't fix it here, but it doesn't matter to me.

**Comment by lrknox** [comment]
He said it had to be in the PR because the line was new for 1_12 except for the PR.

It provided me an opportunity to try committing from a suggestion in the PR on github, which I’d noticed in the spack repo but hadn’t known how to do.  For a minor change (1 missing character in this case) it was much easier than making a change in a clone and pushing it to the fork/branch.

**Comment by byrnHDF** [comment]
So how did you do it, Larry?

________________________________________

**Comment by lrknox** [comment]
*   Select line/lines of code and click ‘+’ for comment.
  *   Click “+/-“ (above “Leave a comment”, to right of “Preview”)
  *   Edit code in suggestion block
  *   “Add single comment” adds suggested change, which has a “commit” button.  I think anyone who has permissions to commit to the branch that the PR is coming from would be able to commit the change.  I assume also that “Start a review” instead of “Add single comment” would also add the suggestion.

**Pull request merged**
